File: Android/app/src/main/java/com/bread/crossboard/model/ClipboardData.kt
package com.bread.crossboard.model

import java.util.UUID

/**
 * Data class for clipboard data
 */
data class ClipboardData(
    val text: String,
    val type: ClipboardType,
    val timestamp: Long,
    var sourceDeviceId: String = "",
    var sourceDeviceName: String = ""
) 

File: Windows/CrossBoard/Models/ClipboardData.cs
using System;

namespace CrossBoard.Models
{
    public enum ClipboardType
    {
        Text,
        Image,
        File
    }

    public class ClipboardData
    {
        public string? Text { get; set; }
        public ClipboardType Type { get; set; }
        public string? SourceDeviceId { get; set; }
        public string? SourceDeviceName { get; set; }
        public long Timestamp { get; set; }
        public int Port { get; set; } = 8765;
    }
} 